### ISSUE TITLE
False positive error for array of labels in matrix for self-hosted runners

### ISSUE THREAD

**Issue by AndreyNautilus** [rhysd/actionlint#164]
I'm using `actionlint` at my work to lint workflows and I encountered an error report which seems to be false-positive.
I have the following (very simplified) workflow:
```yml
on:
  push:
  workflow_dispatch:

jobs:
  build:
    strategy:
      matrix:
        include:
          - msg: hello macos
            host-labels: ["self-hosted", "macOS", "X64"]
          - msg: hello linux
            host-labels: ["self-hosted", "linux"]
    runs-on: ${{ matrix.host-labels }}
    steps:
      - run: echo "${{ matrix.msg }}"
```
which works fine on Github Actions, but `actionlint` reports an error for line `runs-on: ${{ matrix.host-labels }}` (see [the online checker](https://rhysd.github.io/actionlint/#eJx9ULtOAzEQ7O8rRhZlTAFRhPwTFDSRUBT5fI7PsLFPXpsERfn32OIiroFqd2dnZh8xqA6YCo8tnmL6PFA87QfPk86mgt1H7Ln1+uJpaAnAOels3fdPBRx1Tv58rwAfDJXB/gKAxJGdwmiJYuWbyIsmMEbOknRviRXeBVs6yIbZQawgKv/1rSXbzbqFoL9kKiHYJHZ/zSAfyvnfGVk7H5x8MfL5yfWzU7VlGYPCw+Uy3/W4EOJ6nT9gJ77fJ5tKwZoxQix0dZvKF90NaGFmNw==)):

> object, array, and null values should not be evaluated in template with ${{ }} but evaluating the value of type array<string>

The workflow should run the job on 2 different _self-hosted_ runners for which we defined our own labels. According to [Github documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on) `runs-on` supports either a string or a list of string:

> Use `jobs.<job_id>.runs-on` to define the type of machine to run the job on. The machine can be either a GitHub-hosted runner or a self-hosted runner. You can provide `runs-on` as a single string or as an array of strings. If you specify an array of strings, your workflow will run on a self-hosted runner whose labels match all of the specified `runs-on` values, if available.

So, it seems that evaluating an array of string via `${{ }}` should be allowed in this context.

How can I fix/workaround/disable this error for my workflows?

**Comment by AndreyNautilus** [comment]
I found a workaround: first transform `matrix.host-labels` to JSON and then from JSON:
```yaml
runs-on: ${{ fromJSON(toJSON(matrix.host-labels)) }}
```
This works and doesn't trigger lint errors.
However, it's not really convenient to apply such workaround in every workflow that uses self-hosted runners.
